### ISSUE TITLE
Toolbox & sounddevice: Invalid sample rate on playback, microphone not recognized

### ISSUE THREAD

**Issue by KaitoCross** [CorentinJ/Real-Time-Voice-Cloning#94]
I've tried using the toolbox and using the play button throws the following exception on Arch Linux with PulseAudio

```
sounddevice.PortAudioError: Error opening OutputStream: Invalid sample rate [PaErrorCode -9997]
Traceback (most recent call last):
  File "/home/dash/programs/Real-Time-Voice-Cloning/toolbox/__init__.py", line 81, in <lambda>
    func = lambda: self.ui.play(self.ui.selected_utterance.wav, Synthesizer.sample_rate)
  File "/home/dash/programs/Real-Time-Voice-Cloning/toolbox/ui.py", line 142, in play
    sd.play(wav, sample_rate)
  File "/usr/lib/python3.7/site-packages/sounddevice.py", line 154, in play
    **kwargs)
  File "/usr/lib/python3.7/site-packages/sounddevice.py", line 2417, in start_stream
    **kwargs)
  File "/usr/lib/python3.7/site-packages/sounddevice.py", line 1374, in __init__
    **_remove_self(locals()))
  File "/usr/lib/python3.7/site-packages/sounddevice.py", line 780, in __init__
    'Error opening {0}'.format(self.__class__.__name__))
  File "/usr/lib/python3.7/site-packages/sounddevice.py", line 2572, in _check
    raise PortAudioError(errormsg, err)
sounddevice.PortAudioError: Error opening OutputStream: Invalid sample rate [PaErrorCode -9997]
```

Using the recording function throws a similar exception:

`Error opening InputStream: Invalid sample rate [PaErrorCode -9997]`

**Comment by CorentinJ** [comment]
For this I can't help you. You're better off googling that error.

**Comment by 0DARKer** [comment]
Is the problem solved ?

**Comment by robobenklein** [comment]
Do you have more log output?

This seems to be a microphone issue:
```
Expression 'paInvalidSampleRate' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2048
Expression 'PaAlsaStreamComponent_InitialConfigure( &self->capture, inParams, self->primeBuffers, hwParamsCapture, &realSr )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2719
Expression 'PaAlsaStream_Configure( stream, inputParameters, outputParameters, sampleRate, framesPerBuffer, &inputLatency, &outputLatency, &hostBufferSizeMode )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2843
Error opening InputStream: Invalid sample rate [PaErrorCode -9997]
```

while this seems to be an output issue:
```
Expression 'paInvalidSampleRate' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2048
Expression 'PaAlsaStreamComponent_InitialConfigure( &self->playback, outParams, self->primeBuffers, hwParamsPlayback, &realSr )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2722
Expression 'PaAlsaStream_Configure( stream, inputParameters, outputParameters, sampleRate, framesPerBuffer, &inputLatency, &outputLatency, &hostBufferSizeMode )' failed in 'src/hostapi/alsa/pa_linux_alsa.c', line: 2843
Traceback (most recent call last):
  File "/home/robo/code/audio/Real-Time-Voice-Cloning/toolbox/__init__.py", line 81, in <lambda>
    func = lambda: self.ui.play(self.ui.selected_utterance.wav, Synthesizer.sample_rate)
  File "/home/robo/code/audio/Real-Time-Voice-Cloning/toolbox/ui.py", line 142, in play
    sd.play(wav, sample_rate)
  File "/home/robo/code/audio/venv/lib/python3.7/site-packages/sounddevice.py", line 182, in play
    **kwargs)
  File "/home/robo/code/audio/venv/lib/python3.7/site-packages/sounddevice.py", line 2498, in start_stream
    **kwargs)
  File "/home/robo/code/audio/venv/lib/python3.7/site-packages/sounddevice.py", line 1455, in __init__
    **_remove_self(locals()))
  File "/home/robo/code/audio/venv/lib/python3.7/site-packages/sounddevice.py", line 861, in __init__
    'Error opening {0}'.format(self.__class__.__name__))
  File "/home/robo/code/audio/venv/lib/python3.7/site-packages/sounddevice.py", line 2653, in _check
    raise PortAudioError(errormsg, err)
sounddevice.PortAudioError: Error opening OutputStream: Invalid sample rate [PaErrorCode -9997]
```

(and I have both of them... yay...)

Seems to be a problem with devices on PulseAudio not using a default sample rate that sounddevice expects. I'll try either changing sample rates / resolution in pulseaudio or diving into sounddevice's code to see if it's possible to correct the samplerate detection. (41 vs 48 khz?)

Ubuntu/Pop_OS, pulseaudio 12.2-2, Default Sample Specification: `s24le 2ch 48000Hz`  
Microphone: s16le 1ch 48000Hz  
Output: float32le 2ch 48000Hz / s32le 2ch 48000Hz

**Comment by rmaria** [comment]
It's not a microphone problem. Try to use something else than sounddevice to play the output audio and it will work. Instead of sd.stop() and sd.play(generated_wav, synthesizer.sample_rate) in demo_cli.py use scipy.io.wavfile

**Comment by blue-fish** [comment]
This issue is resolved now that #390 is merged. Please pull the latest version of the code and try again.
